File: oj-web/src/enums/oj/judge/JudgeSceneType.ts
import type {BaseEnum} from '@/enums/base';

// 评测场景类型枚举接口
interface JudgeSceneTypeEnum extends BaseEnum {
    tooltip?: string;
    text?: string;
}

/**
 * 评测场景类型枚举
 * 用于标识不同的评测场景
 */
export const JudgeSceneType: { [key: string]: JudgeSceneTypeEnum } = {
    Normal: {
        code: 'Normal',
        name: '普通测评',
        tooltip: '普通评测场景',
        text: '常规代码评测场景'
    },
    Exercise: {
        code: 'Exercise',
        name: '练习',
        tooltip: '练习评测场景',
        text: '用于练习的评测场景'
    },
    Homework: {
        code: 'Homework',
        name: '作业',
        tooltip: '训练作业评测场景',
        text: '用于日常作业训练的评测场景'
    },
    Contest: {
        code: 'Contest',
        name: '比赛',
        tooltip: '比赛评测场景',
        text: '用于比赛的评测场景'
    }

};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): JudgeSceneTypeEnum | null {
    return Object.values(JudgeSceneType).find(item => item.code === code) || null;
}

File: glowxq-api/business/business-oj/src/main/java/com/glowxq/oj/judge/enums/JudgeSceneType.java
package com.glowxq.oj.judge.enums;

import com.glowxq.core.common.enums.base.BaseType;
import com.glowxq.oj.topic.enums.TopicType;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.List;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum JudgeSceneType implements BaseType {

    Normal("Normal", "普通测评"),
    Exercise(TopicType.Exercise.getCode(), TopicType.Exercise.getName()),
    Homework(TopicType.Homework.getCode(), TopicType.Homework.getName()),
    Contest(TopicType.Contest.getCode(), TopicType.Contest.getName()),
    ;

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static JudgeSceneType matchCode(String code) {
        for (JudgeSceneType sceneType : JudgeSceneType.values()) {
            if (sceneType.getCode().equals(code)) {
                return sceneType;
            }
        }
        return null;
    }

    public TopicType toTopicType() {
        return TopicType.matchCode(this.getCode());
    }

    public Boolean isFixedTopic(){
        return List.of(Exercise, Homework, Contest).contains(this);
    }
}
